### PR TITLE
Java cleanup

### PR DESCRIPTION
**Pull request by jsoref**
This is for discussion.

Notes:
* Most commits were created by IntelliJ (the descriptions are mine because I lost the details)
* a number of members aren't tagged as `final` but appear to be designed to be mutated by a script or debugger -- I've tried to leave those alone. Once these `final`s are added, I can go back and add comments warning people that the others are intentional (some already have a hint, but not all)
* Individual commits are mostly self contained (except for whitespace cleanup which precedes some other change)
* In general, most commits should not change program behavior at all
* 2842b43 is a manual attempt to address non-atomic access to a volatile variable (not sure if it works)
* 78f5fe7 is a behavior change for what appears to be a bug, as is, the function is an expensive NOP

* Calculating the right `serialVersionUID` is painful, I mostly used this:
```
CLASSPATH=$(echo ~/.jenkins-2.161/war/WEB-INF/lib/*.jar|tr ' ' ':')
for a in `cat`; do serialver -classpath $CLASSPATH $a 2>/dev/null; done|grep 'serialVersionUID'|egrep [a-z]
````
But it didn't always give an answer. I've tried to flag all those instances.

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: core/src/main/java/hudson/ClassicPluginStrategy.java [hunk 1]
```diff
@@ -705,5 +706,5 @@ protected Class defineClassFromData(File container, byte[] classData, String cla
     }
 
     public static boolean useAntClassLoader = SystemProperties.getBoolean(ClassicPluginStrategy.class.getName()+".useAntClassLoader");
-    public static boolean DISABLE_TRANSFORMER = SystemProperties.getBoolean(ClassicPluginStrategy.class.getName()+".noBytecodeTransformer");
+    public static final boolean DISABLE_TRANSFORMER = SystemProperties.getBoolean(ClassicPluginStrategy.class.getName()+".noBytecodeTransformer");
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/ClassicPluginStrategy.java]: This is not recommended. We want properties to be controllable from System Groovy scripts if needed

File: core/src/main/java/hudson/Platform.java [hunk 2]
```diff
@@ -47,7 +47,7 @@
      */
     public final char pathSeparator;
 
-    private Platform(char pathSeparator) {
+    Platform(char pathSeparator) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/Platform.java]: why?

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thanks @jsoref !

System Properties static vars will need to be reverted. The rest looks OK, but it will likely cause  issues with merge conflicts and with backporting of fixes. Bulk cleanup of code is still a good change IMO, but we need to sync-up with @olivergondza and @daniel-beck 

Also, it might be easier to have smaller PRs with single refactoring categories

File: core/src/main/java/hudson/Platform.java (hunk 2 above)
Comment (jsoref) [reply core/src/main/java/hudson/Platform.java]: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html
Note: The constructor for an enum type must be package-private or private access. It automatically creates the constants that are defined at the beginning of the enum body. You cannot invoke an enum constructor yourself.

File: core/src/main/java/hudson/ClassicPluginStrategy.java (hunk 1 above)
Comment (jsoref) [reply core/src/main/java/hudson/ClassicPluginStrategy.java]: Oops, I tried to drop all of those (as noted above)... This wasn't intended.

**Comment by jsoref** [comment]
Yes, we can certainly split this. Select the commits by subject and I'll split them. I don't think 50+ PRs is the right choice either.

File: core/src/main/java/hudson/Platform.java (hunk 2 above)
Comment (jvz) [reply core/src/main/java/hudson/Platform.java]: Redundant keyword, just like `public` inside an interface, or static inner classes of an interface, or static inner enums (all redundant modifiers).

**Comment by jvz** [comment]
This is awesome that you're doing this. These small little messes have been rather annoying.

**Comment by jsoref** [comment]
Can someone guess what's going wrong here? My guess is that it's the serialversion stuff (when I return to working on this, I'll try splitting that out).

File: core/src/main/java/hudson/util/ChunkedInputStream.java [hunk 3]
```diff
@@ -342,9 +343,8 @@ public void close() throws IOException {
      */
     static void exhaustInputStream(InputStream inStream) throws IOException {
         // read and discard the remainder of the message
-        byte buffer[] = new byte[1024];
+        byte[] buffer = new byte[1024];
         while (inStream.read(buffer) >= 0) {
-            ;
```
Comment (alecharp) [inline core/src/main/java/hudson/util/ChunkedInputStream.java]: maybe this was to comply with findbugs

**Review by alecharp — APPROVED** [review]
I don't know why the build is failing. @jsoref could you please close, wait 1min and reopen your PR to trigger a new build?

**Review by olivergondza — APPROVED** [review]
Looks good to me!

**Comment by alecharp** [comment]
@oleg-nenashev can you have a new look at this please?

**Review by oleg-nenashev — APPROVED** [review]

File: core/src/main/java/hudson/util/ChunkedInputStream.java (hunk 3 above)
Comment (jsoref) [reply core/src/main/java/hudson/util/ChunkedInputStream.java]: I'm not sure, I'm pretty sure we're inconsistent. I think that if we're trying to comply w/ a given tool we should always include a comment to help others know that.

**Comment by jsoref** [comment]
Note: this is something like half of the changes I had earlier because I can't figure out why CI doesn't like me. If this is green and merged, I'll make a new PR for another portion and we'll iterate until it's done.

**Pull request merged**